src/NuGet.Core/NuGet.ProjectModel/PackageSpecExtensions.cs

```diff
@@ -24,6 +25,13 @@ public static TargetFrameworkInformation GetTargetFramework(this PackageSpec pro
             return frameworkInfo ?? new TargetFrameworkInformation();
         }
 
+        public static TargetFrameworkInformation GetTargetFramework(this PackageSpec project, string targetAlias)
+        {
+            var frameworkInfo = project.TargetFrameworks.FirstOrDefault(f => targetAlias.Equals(f.TargetAlias, StringComparison.OrdinalIgnoreCase));
+
+            return frameworkInfo ?? new TargetFrameworkInformation();
```

Comment:
What's a scenario where a target alias is defined, but doesn't exist in the package spec?

I know NuGet has a history of silently ignoring errors, but should we be throwing a `KeyNotFoundException`, or something like that?  If the API caller requests a target alias that was not defined in the csproj, I think it should throw an exception.  But if it's possible that the targetAlias is defined in the csproj, but for some other reason is not in the packageSpec, then I'd find this more acceptable.

Reply:
I think you are right. 
I'll remove the extension method, or rather I'll make it internal to NuGet.Commands and make it fail fast.